ts/mains/main_node.ts

```diff
@@ -1226,6 +1230,66 @@ ipc.on('media-access', async () => {
   await askForMediaAccess();
 });
 
+// Proxy settings
+let proxyUsername: string | null = null;
+let proxyPassword: string | null = null;
+
+async function applyProxySettings() {
+  try {
+    const enabled = sqlNode.getItemById('proxy-enabled')?.value || false;
+
+    if (!enabled) {
+      // Clear proxy if disabled
+      await session.defaultSession.setProxy({ proxyRules: '' });
+      console.log('Proxy disabled');
+      return;
+    }
+
+    const host = (sqlNode.getItemById('proxy-host')?.value || '') as string;
+    const port = sqlNode.getItemById('proxy-port')?.value || 0;
+    const username = (sqlNode.getItemById('proxy-username')?.value || '') as string;
+    const password = (sqlNode.getItemById('proxy-password')?.value || '') as string;
+
+    if (!host || !port) {
+      console.warn('Proxy enabled but host or port is missing');
+      return;
+    }
+
+    // Store credentials for login handler
+    proxyUsername = username || null;
+    proxyPassword = password || null;
+
+    const proxyRules = `socks5://${host}:${port}`;
+    await session.defaultSession.setProxy({
+      proxyRules,
+      proxyBypassRules: '<local>',
+    });
+
+    console.log(`Proxy configured: ${proxyRules}`);
+  } catch (e) {
+    console.error('Failed to apply proxy settings:', e);
+  }
+}
+
+// Handle proxy authentication
+app.on('login', (_event, _webContents, _authenticationResponseDetails, authInfo, callback) => {
+  if (authInfo.isProxy && proxyUsername && proxyPassword) {
+    callback(proxyUsername, proxyPassword);
+  } else {
+    callback('', ''); // Cancel the auth request
+  }
+});
```

Comment:
can you remove proxyUsername & password from the file and instead fetch those here on the login callback?

i.e.

```suggestion
app.on('login', (_event, _webContents, _authenticationResponseDetails, authInfo, callback) => {
  const username = (sqlNode.getItemById('proxy-username')?.value || '') as string;
  const password = (sqlNode.getItemById('proxy-password')?.value || '') as string;
  if (authInfo.isProxy && username && password) {
    callback(proxyUsername, proxyPassword);
  } else {
    callback('', ''); // Cancel the auth request
  }
});
```